concordium-consensus/src/Concordium/GlobalState/Basic/BlockState.hs

```diff
@@ -556,6 +556,56 @@ doGetActiveBakersAndDelegators bs = return (bakers, passiveDelegatorInfos)
     passive = Set.toAscList $ bs ^. blockBirkParameters . birkActiveBakers . passiveDelegators . apDelegators
     !passiveDelegatorInfos = mkActiveDelegatorInfo <$> passive
 
+doGetActiveDelegators ::
+    ( HasBlockState s pv,
+      IsProtocolVersion pv,
+      Monad m,
+      AccountVersionFor pv ~ 'AccountV1
+    ) =>
+    s ->
+    Maybe BakerId ->
+    m (Maybe [(AccountAddress, BS.ActiveDelegatorInfo)])
+doGetActiveDelegators bs mPoolId =
+  case mPoolId of
+    Nothing -> return (Just passiveDelegatorInfos)
+    Just poolId ->
+      case bs ^. blockBirkParameters . birkActiveBakers . activeBakers . at poolId of
+        Nothing -> return Nothing
+        Just ActivePool{..} -> return . Just . fmap mkActiveDelegatorInfo . Set.toAscList $ _apDelegators
+  where
+    mkActiveDelegatorInfo activeDelegatorId@(DelegatorId acct) =
+        (addr, BS.ActiveDelegatorInfo
+            { activeDelegatorStake = theDelegator ^. delegationStakedAmount,
+              activeDelegatorPendingChange = theDelegator ^. delegationPendingChange,
+              ..
+            })
+      where
+        !theDelegator = bs ^. blockAccounts . Accounts.unsafeIndexedAccount acct . unsafeAccountDelegator
+        !addr = bs ^. blockAccounts . Accounts.unsafeIndexedAccount acct . accountAddress
```

Comment:
Should we add a comment about the use of apparently unsafe functions here?

Reply:
I'll leave this for another time, since this is for testing only and we plan to restructure this anyway.